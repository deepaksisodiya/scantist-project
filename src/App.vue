<template>
  <div id="app">
    <HerosTable />
  </div>
</template>

<script>
import HerosTable from "./components/HerosTable.vue";

export default {
  name: "app",
  components: {
    HerosTable
  }
};
</script>

<style>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
