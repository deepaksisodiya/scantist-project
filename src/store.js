import Vue from "vue";
import Vuex from "vuex";
import axios from "axios";

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    isLoading: false,
    isError: false,
    heros: []
  },

  mutations: {
    SET_LOADING_STATUS(state, isLoading) {
      state.isLoading = isLoading;
    },
    SET_ERROR_STATUS(state, isError) {
      state.isError = isError;
    },
    SET_HEROS(state, heros) {
      state.heros = heros;
    }
  },

  actions: {
    async fetchHeros(context) {
      context.commit("SET_LOADING_STATUS", true);
      context.commit("SET_ERROR_STATUS", false);

      try {
        const response = await axios.get(
          "https://api.opendota.com/api/heroStats"
        );
        if (response.status === 200) {
          context.commit("SET_HEROS", response.data);
        }
      } catch (error) {
        context.commit("SET_ERROR_STATUS", true);
      } finally {
        context.commit("SET_LOADING_STATUS", false);
      }
    }
  },

  getters: {
    filterHerosByName: state => searchName => {
      return state.heros.filter(hero => {
        const name = hero.localized_name.toLowerCase();
        const search = searchName.toLowerCase();
        return name.includes(search);
      });
    }
  }
});
