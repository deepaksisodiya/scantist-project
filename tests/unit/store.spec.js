import { mutations, getters } from "../../src/store";

describe("Store.js", () => {
  test("set loading state to true", () => {
    const state = {
      isLoading: false,
      isError: false,
      heros: []
    };
    mutations.SET_LOADING_STATUS(state, true);
    expect(state.isLoading).toBe(true);
  });

  test("set error state to true", () => {
    const state = {
      isLoading: false,
      isError: false,
      heros: []
    };
    mutations.SET_ERROR_STATUS(state, true);
    expect(state.isError).toBe(true);
  });

  test("set heros to some value", () => {
    const state = {
      isLoading: false,
      isError: false,
      heros: []
    };
    mutations.SET_HEROS(state, [
      { id: 1, localized_name: "hero1" },
      { id: 2, localized_name: "hero2" }
    ]);
    expect(state.heros).toMatchSnapshot();
  });

  test("testing getters for filterHerosByName", () => {
    const state = {
      heros: [
        { id: 1, localized_name: "Deepak" },
        { id: 2, localized_name: "Piyush" },
        { id: 1, localized_name: "Mayur" },
        { id: 1, localized_name: "Ankur" }
      ]
    };
    const filterHeros = getters.filterHerosByName(state)("dee");
    expect(filterHeros).toMatchSnapshot();
    expect(filterHeros.length).toBe(1);
  });
});
