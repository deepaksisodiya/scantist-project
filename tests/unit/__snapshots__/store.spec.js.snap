// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store.js set heros to some value 1`] = `
Array [
  Object {
    "id": 1,
    "localized_name": "hero1",
  },
  Object {
    "id": 2,
    "localized_name": "hero2",
  },
]
`;

exports[`Store.js testing getters for filterHerosByName 1`] = `
Array [
  Object {
    "id": 1,
    "localized_name": "Deepak",
  },
]
`;
